src/System.Diagnostics.Process/tests/System.Diagnostics.Process.Tests/ProcessTests.cs

```diff
@@ -187,36 +187,31 @@ public void TestMachineName()
             Assert.NotNull(_process.MachineName);
         }
 
-        [Fact]
+        [Fact, PlatformSpecific(~PlatformID.OSX)]
         public void TestMainModule()
         {
+            // Ensure the process has loaded the modules.
+            while (_process.Modules.Count == 0) ;
```

Comment:
> I guess the individual test method would timeout after a point even if it did, right?

I don't believe so. If we really need to do this kind of spinning, it'd be better to do it something like:
```C#
Assert.True(SpinWait.SpinUntil(() => _process.Modules.Count == 0, WaitInMS));
```